src/main/java/com/olgaboiar/mint/Router.java

```diff
@@ -3,43 +3,57 @@
 import com.olgaboiar.mint.handlers.*;
 
 import java.io.IOException;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
 
 public class Router {
     private Map<String, Map<String, IHandler>> routes = new HashMap<>();
+//    Map<String, IHandler> methodHandlers;
 
     public Router() {
         registerTestRoutes();
     }
 
     public Response route(Request request) throws IOException {
-        Map<String, IHandler> methodList = routes.get(request.getRequestedFile());
-        if (methodList == null) {
+        Map<String, IHandler> methodHandlers = routes.get(request.getUri());
+        if (methodHandlers == null) {
             return new NotFoundHandler().handleRequest(request);
         }
-        Set<String> allowedMethods = methodList.keySet();
-        IHandler handler =  methodList.getOrDefault(request.getMethod(), new NotAllowedHandler(allowedMethods));
+        String[] allowedMethods = new String[methodHandlers.keySet().size()];
+        allowedMethods = methodHandlers.keySet().toArray(allowedMethods);
+        IHandler handler =  methodHandlers.getOrDefault(request.getMethod(), new NotAllowedHandler(allowedMethods));
         return handler.handleRequest(request);
     }
 
-    public void setHandler(Route route, IHandler routeHandler) {
-        Map<String, IHandler> methodList = routes.get(route.getPath());
-        if(methodList == null) {
-            methodList = new HashMap();
-            routes.put(route.getPath(), methodList);
+    public void setMethodHandlers(Route route) {
+        Map<String, IHandler> methodHandlers = routes.get(route.getPath());
+        if(methodHandlers == null) {
+            methodHandlers = new HashMap();
+            routes.put(route.getPath(), methodHandlers);
+        }
+        String[] allowedForThisPath = route.getAllowedMethods();
+        for (String method : allowedForThisPath) {
+            methodHandlers.put(method, setHandler(method, allowedForThisPath));
         }
-        methodList.put(route.getMethod(), routeHandler);
+    }
+
+    public IHandler setHandler(String method, String[] allowedForThisPath) {
+        switch (method) {
+            case "GET":
+                return new RouteHandler();
+            case "HEAD":
+                return new HeadHandler();
+            case "OPTIONS":
+                return new OptionsHandler(allowedForThisPath);
+        }
+        return new NotAllowedHandler(allowedForThisPath);
     }
 
     public void registerTestRoutes() {
```

Comment:
Think about how the user would be able to use the router, for example, setting new routes? Currently, all the routes are predefined within your server and there's no way to add new routes. Is there a way to enable this?

Reply:
fixed in https://github.com/olgaboiar/MintHTTPServer/pull/9/commits/271647c8faa7184a094aa074a376fc45d485bee6

Reply:
Nice work refactoring that out! Something to keep in mind: as a user, if I'm hosting my app on your server, how would I add resources or add endpoints to your router without touching your server code?